File: app/iOS/Thermal Comfort Study/MinuteCaloriesItem.swift
//
//  MinuteCaloriesItem.swift
//  Thermal Comfort Study
//
//  Created by Madhav Achar on 10/14/16.
//  Copyright © 2016 boschsc. All rights reserved.
//

import Foundation

class MinuteCaloriesItem: BandDataItem {
    
    private static let MINUTE_CALORIES_NAME = "MinuteCalories"
    private var minuteCalories: Double!
    
    init(username: String, timestamp: Date, minuteCalories: Double) {
        super.init(username, timestamp)
        self.minuteCalories = minuteCalories
    }
    
    func toXML() -> String {
        return super.toXML(dataName: type(of: self).MINUTE_CALORIES_NAME,
                           value: self.minuteCalories)
    }
    
    
}

File: app/android/app/src/main/java/bosch/smartcampus/thermalcomfortstudy/packet/MinuteCaloriesItem.java
package bosch.smartcampus.thermalcomfortstudy.packet;

import bosch.smartcampus.thermalcomfortstudy.lib.Timestamp;

/**
 * Created by rsukkerd on 6/27/16.
 */
public class MinuteCaloriesItem extends BandDataItem {
    private static final String MINUTE_CALORIES_NAME = "MinuteCalories";

    private double minuteCalories;

    public MinuteCaloriesItem(String username, Timestamp timestamp,
                              double minuteCalories) {
        super(username, timestamp);
        this.minuteCalories = minuteCalories;
    }

    public double getMinuteCalories() {
        return minuteCalories;
    }

    @Override
    public String toXML() {
        StringBuilder builder = new StringBuilder();
        builder.append("<" + ELEMENT + " id=\"" + getId() + "\">");
        builder.append(getTransducerDataElement(MINUTE_CALORIES_NAME, Double.toString(minuteCalories)));
        builder.append("</" + ELEMENT + ">");
        return builder.toString();
    }
}
